Harmony/Public/Attributes.cs

```diff
@@ -18,7 +18,11 @@ public enum MethodType
 		/// <summary>This is a static constructor</summary>
 		StaticConstructor,
 		/// <summary>This targets the MoveNext method of the enumerator result</summary>
-		Enumerator
+		Enumerator,
+#if NET40_OR_GREATER
+		/// <summary>This targets the MoveNext method of the async state machine that actually has the method's implementation</summary>
```

Comment:
If you shorten the comment a bit (same style as AccessTools.cs:504) I'll approve the PR